docs/io.cozy.contacts.md

```diff
@@ -6,6 +6,8 @@
 
 The `io.cozy.contacts` doctype is loosely based on the [vCard RFC](https://tools.ietf.org/html/rfc6350), in that some of the attributes have been renamed for clarity. The attributes with a `?` are optional.
 
+- `index` : {string} Field to be used to sort contacts. It's a concatenation of `givenName`, `familyName`, first `email` and `cozy.url`
```

Comment:
I don't think `index` is a good name, it's going to be confusing to set and index on `index`. An it's just one of possible indexes.

How about `sort` or `sortField` ?

Reply:
yes I proposed `sortBy`too. Ok for `sort` 👍

Reply:
I'm wondering if we should not create something more robust over time. 

I don't know exactly yet, but having something like : 
```
indexes: {
 byGivenNameFamilyNameEmailCozyUrl: value
}
```
seems better to me.

Reply:
Est-ce qu'on commencerait pas par faire simple, pour complexifier ensuite si besoin ? (lean)

Reply:
It's always a bit of work to migrate datas... We need to think of a fairly generic structure right from the start.

Reply:
In that case, maybe `alphabeticalSort` ? It's more specific than `sort` and leaves us some options if we want other sorts, without being overly specific

Reply:
Since it's not a "primary" data, I tend to think that we should not put it in the root of the document. Having an `indexes` object / array looks like a good idea. 

And since this entry is pretty special, having its name speaking for itself is mandatory. `alphabeticalSort ByGivenNameFamilyNameEmailCozyUrl` ?

Reply:
Is `alphabeticalSort` relevant here ? I mean, this will be part of the query (how to sort), not of how we create and store the value of this attribute (what to store).

And to create its value, we decided to concatenate givenName + familyName + mail + url but is it not simpler to get the first not empty of those fields ?

So go for this way : 
```
root {
   indexes: {
      ByGivenNameFamilyNameEmailCozyUrl: value
   }
}
```
?

Reply:
File change according to comments above

Reply:
Good point on the alphabetical sort not being too relevant here. 👍 for the suggestion above.